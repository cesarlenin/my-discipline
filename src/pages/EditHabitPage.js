import React, { Component } from 'react'
import EditHabitForm from '../components/EditHabitForm';
import UserContext from '../components/UserContext'

export class EditHabitPage extends Component {
  static contextType = UserContext;
  render() {
    const { habits } = this.context;
    const shownHabit = habits.find(
      (habit) => habit.id === this.props.match.params.habitId
    );
    return (
      <main>
      <header>
        <h1>Edit Habit</h1>
      </header>
      <section>
          <EditHabitForm habit={shownHabit} />
      </section>
    </main>
    )
  }
}

export default EditHabitPage