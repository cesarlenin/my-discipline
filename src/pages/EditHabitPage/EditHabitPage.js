import React, { Component } from 'react';
import EditHabitForm from '../../components/EditHabitForm/EditHabitForm';
import UserContext from '../../components/UserContext/UserContext'
import BackButton from '../../components/BackButton/BackButton';
import './EditHabitPage.css';

export class EditHabitPage extends Component {
  static contextType = UserContext;
  render() {
    const {
      habits
    } = this.context;
    
    const shownHabit = habits.find(
      (habit) => habit.id === Number(this.props.match.params.habitId)
    );
    return ( 
      <section >
        <header >
          <h1 > Edit Habit </h1> 
        </header> 
        <BackButton {...this.props}/> 
        <section className = 'edit' >
          <EditHabitForm habit = {shownHabit} {...this.props}/> 
        </section> 
      </section>
    )
  }
}

export default EditHabitPage;