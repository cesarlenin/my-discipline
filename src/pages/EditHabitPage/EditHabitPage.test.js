import React from 'react';
import ReactDOM from 'react-dom';
import EditHabitPage from './EditHabitPage';
import store from '../../dummy-store';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <EditHabitPage
  habit={store.habits[0]}
  />, div);
  ReactDOM.unmountComponentAtNode(div);
});