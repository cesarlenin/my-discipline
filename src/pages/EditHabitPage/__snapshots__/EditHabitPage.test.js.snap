// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditHabitPage /> renders this UI as expected 1`] = `
<section>
  <header>
    <h1>
      Edit Habit
    </h1>
  </header>
  <div
    className="back"
  >
    <button
      onClick={[Function]}
    >
      Go Back
    </button>
  </div>
  <section
    className="edit"
  >
    <form
      className="edit-habit-form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="name"
      >
        Name:
      </label>
      <input
        defaultValue="Read a book"
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="description"
      >
        Description:
      </label>
      <input
        defaultValue="Important"
        id="description"
        name="description"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="Goal"
      >
        Goal:
      </label>
      <input
        defaultValue={7}
        id="Goal"
        max="7"
        min="1"
        name="Goal"
        onChange={[Function]}
        type="number"
      />
      <button
        type="submit"
        value="Submit"
      >
        Submit
      </button>
    </form>
  </section>
</section>
`;
